econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -715,15 +745,27 @@ def fit(self, X, y, warm_start=False):
         # assume we'll be able to train former failures this time; we'll add them back if not
         invalid_inds = [(ind, reason) for (ind, reason) in invalid_inds if ind not in new_inds]
 
+        self._has_column_names = True
```

Comment:
Why are these lines moved up here?

Reply:
This was necessary so that `_format_col` could be used to produce the warning messages below, since we need to have access to the column names in order to decide how to format the column